---
title: "GOSeq"
author: "LaurenZane"
date: "6/2/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## load relevant packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet") 
library("goseq")
library("gridExtra")
library("clusterProfiler")
library("DataCombine")
library("VennDiagram")
```
## load gene count matrix, gene length dataframe, and DESeq2 significant results dataframe
```{r}
## filtered gene count matrix pOverA (0,5,1) from DESeq script

gcount_filt <- read.csv("/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Output/Pacuta/20220608_Pacuta_filtered_gene_count.csv") # gene counts filtered for PoverA(0.5,10)
colnames(gcount_filt) # data frame does not currently have a true column for gene_id
#gcount_filt <- tibble::rownames_to_column(gcount_filt, "gene_id") # create true column for gene_id
#gcount_filt <- subset (gcount_filt, select = -X)
colnames(gcount_filt)[1] <- "gene_id"
colnames(gcount_filt)
rownames(gcount_filt) <- gcount_filt$gene_id #assign row names to gene_id 
print(rownames(gcount_filt))

gcount_filt$gene_id <- gsub("\\_t*$","", as.character(gcount_filt$gene_id))# removes _t from gene id
gcount_filt$gene_id <- gsub("t1","", as.character(gcount_filt$gene_id))# removes t1 from gene id
gcount_filt$gene_id <- gsub("t2","", as.character(gcount_filt$gene_id))# removes t2 from gene id
gcount_filt$gene_id <- gsub("t3","", as.character(gcount_filt$gene_id))# removes t3 from gene id
gcount_filt$gene_id <- gsub("t4","", as.character(gcount_filt$gene_id))# removes t4 from gene id
gcount_filt$gene_id <- gsub("\\.*$","", as.character(gcount_filt$gene_id)) # removes . from end of gene id
head(gcount_filt$gene_id)

# gene length data frame

gene_lengths <- read.csv("/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Data/Pacuta/length.mRNA_Pacuta.csv", row.names = NULL) # gene length information 
gene_lengths$gene_id <- gsub("\\_t*$","", as.character(gene_lengths$gene_id))# removes t1 from gene id
gene_lengths$gene_id <- gsub("t1","", as.character(gene_lengths$gene_id))# removes t1 from gene id
gene_lengths$gene_id <- gsub("t2","", as.character(gene_lengths$gene_id))# removes t2 from gene id
gene_lengths$gene_id <- gsub("t3","", as.character(gene_lengths$gene_id))# removes t3 from gene id
gene_lengths$gene_id <- gsub("t4","", as.character(gene_lengths$gene_id))# removes t4 from gene id
gene_lengths$gene_id <- gsub("\\.*$","", as.character(gene_lengths$gene_id)) # removes . from end of gene id
print(gene_lengths$gene_id)
unique_gene_lengths <- unique(gene_lengths[1])
unique_gene_lengths$gene_id == gene_lengths$gene_id
gene_length.keep <- rownames(unique_gene_lengths) # create a list of unique gene id's from gene length df
class(gene_length.keep) 
gene_length_final <- as.data.frame(gene_lengths[which(rownames(gene_lengths) %in% gene_length.keep),])# create a df that only includes unique gene ids 
rownames(gene_length_final) <- gene_length_final$gene_id #assign row names to gene_id 
print(rownames(gene_length_final)) #confirm that rownames are gene id's

# DEG datframe from DESeq script 
DEG <- read.csv("~/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Output/Pacuta/20220608_PacutaDEGSeq2.sig.results.csv", row.names=NULL) # significantly differentially expressed genes from DESeq2 results (16285 genes)
colnames(DEG) # DEG df already has a true column for gene id
DEG <- subset (DEG, select = -gene_id)
colnames(DEG)[1] <- "gene_id"
rownames(DEG) <- DEG$gene_id #assign row names to gene_id 
print(rownames(DEG))
DEG$gene_id <- gsub("\\_t*$","", as.character(DEG$gene_id))# removes t1 from gene id
DEG$gene_id <- gsub("t1","", as.character(DEG$gene_id))# removes t1 from gene id
DEG$gene_id <- gsub("t2","", as.character(DEG$gene_id))# removes t2 from gene id
DEG$gene_id <- gsub("t3","", as.character(DEG$gene_id))# removes t3 from gene id
DEG$gene_id <- gsub("t4","", as.character(DEG$gene_id))# removes t4 from gene id
DEG$gene_id <- gsub("\\.*$","", as.character(DEG$gene_id)) # removes . from end of gene id
print(DEG$gene_id)
unique_DEG <- unique(DEG)
colnames(unique_DEG)
rownames(DEG)

```
## check to make sure that the gene id's for all data frames are the SAME. This is essential for creating the vectors for GOSeq
```{r}
length(intersect(gcount_filt$gene_id, gene_lengths$gene_id)) # go back to previous code block and remove ".tl,", ".t2", ".t3", "_t" from end of all predicted genes then rerun this chunk
length(intersect(gcount_filt$gene_id, DEG$gene_id))
```
## load GO term df
```{r}
GO.terms <- read.csv("/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Data/Pacuta/Pacuta_GOterms_20220121.csv")
colnames(GO.terms) # check column names, need to remove "X" and change "SeqName" column to "gene_id"
GO.terms <- subset (GO.terms, select = -X)
colnames(GO.terms) <- c("gene_id","GO.ID")
colnames(GO.terms) # check column names
GO.terms$gene_id <- gsub("\\_t*$","", as.character(GO.terms$gene_id))# removes t1 from gene id
GO.terms$gene_id <- gsub("t1","", as.character(GO.terms$gene_id))# removes t1 from gene id
GO.terms$gene_id <- gsub("t2","", as.character(GO.terms$gene_id))# removes t2 from gene id
GO.terms$gene_id <- gsub("t3","", as.character(GO.terms$gene_id))# removes t3 from gene id
GO.terms$gene_id <- gsub("t4","", as.character(GO.terms$gene_id))# removes t4 from gene id
GO.terms$gene_id <- gsub("\\.*$","", as.character(GO.terms$gene_id)) # removes . from end of gene id

GO.terms$GO.ID<- as.character(GO.terms$GO.ID)
#GO.terms$GO.ID <- replace_na(GO.terms$GO.ID, "unknown")
GO.terms$GO.ID <- as.factor(GO.terms$GO.ID)
GO.terms$gene_id <- as.factor(GO.terms$gene_id)
GO.terms$GO.ID <- gsub(" ", "", GO.terms$GO.ID)
GO.terms <- unique(GO.terms)
```
# merge gene length dataframe with filtered counts as well as DEG df by gene_id. This ensures that gene id's are consistent across dataframes 
```{r}
gcount_filt_length_merge <- merge(gcount_filt, gene_length_final, by = "gene_id")
dim(gcount_filt_length_merge)
colnames(gcount_filt_length_merge)
unique_length_merge <- unique(gcount_filt_length_merge[1]) # 684
unique_length_merge2 <- rownames(unique_length_merge)
length_merge_final <- as.data.frame(gcount_filt_length_merge[which(rownames(gcount_filt_length_merge) %in% unique_length_merge2),])# create a df that only includes unique gene ids 
dim(length_merge_final) # 684 x 8
# update 20220608


```
## create GO Seq vectors from all genes, DEGs, and gene lengths
```{r}
## ID vector 

IDvector <- length_merge_final$gene_id
is.vector(IDvector) # needs to be TRUE
length(IDvector)
# 684
# updated 23054
length(unique(IDvector))
# 684

## DEG vector

DEG_vector <- length_merge_final[length_merge_final$gene_id %in% DEG$gene_id, ]
dim(DEG_vector) # 185 
# updated 6068
DEG_names <- as.vector(DEG$gene_id)
gene_vector <- as.integer(length_merge_final$gene_id%in%DEG_names)
names(gene_vector) <- length_merge_final$gene_id
length(gene_vector)
# 684

## gene length vector 

length.vector <- length_merge_final$length
class(length.vector) # currently integer
length.vector <- as.numeric(length.vector) # convert to numeric
class(length.vector)
is.vector(length.vector)# TRUE, is a vector
length(length.vector)

# 684
# 23054 updated 20220608

head(gene_vector)
print(gene_vector)
head(length.vector)
head(IDvector)
```
## calculate probability weighing function
```{r}
# genome <- read.csv("/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Data/Pacuta/Pacuta_annot_20220121.csv")

library("goseq")
?nullp
#names(length.vector)
# arguments
# DEgenes, named binary vector w/ DEGs as 1 and others as 0
# genome: string identifying genome, is this option totally necessary?
# id: string identifying gene IDs


length(gene_vector) == length(length.vector) # confirm that length and gene vector are the same length, pwf will not work without it!
pwf = nullp(gene_vector, IDvector, bias.data = length.vector)

```
## perform GOSeq
```{r}
?goseq
GO.wall<-goseq(pwf, IDvector, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE) # perform GOSeq
write.csv(GO.wall, file = "/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Output/Pacuta/GOSeq/20220608_GO.wall.csv") # write csv to save to Output
```
## find significantly enriched GO terms
```{r}
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")
enriched.GO.05 <- enriched.GO.05[order(-enriched.GO.05$numDEInCat),]
enriched.GO.05$term <- as.factor(enriched.GO.05$term)
head(enriched.GO.05)

# Subset enriched GO terms by ontology (BP, CC, MF) and save as csv. 
MF <- subset(enriched.GO.05, ontology=="MF")
MF <- MF[order(-MF$numDEInCat),]
CC <- subset(enriched.GO.05, ontology=="CC") 
CC <- CC[order(-CC$numDEInCat),] 
BP <- subset(enriched.GO.05, ontology=="BP")
BP <- BP[order(-BP$numDEInCat),]

write.csv(MF, file = "/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Output/Pacuta/GOSeq/Pacuta_MF_Sig_Enriched_GO_20220608.csv")
write.csv(CC, file = "/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Output/Pacuta/GOSeqPacuta_CC_Sig_Enriched_GO_20220608.csv")
write.csv(BP, file = "/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Output/Pacuta/GOSeq/Pacuta_BP_Sig_Enriched_GO_20220608.csv")
write.csv(enriched.GO.05, file = "/Users/laurenzane/Desktop/Putnam_Lab/Projects/Express_Compare/RAnalysis/Output/Pacuta/GOSeq/Pacuta_Sig_Enriched_GO_20220608.csv")
```

