## stringtie assembly for *P acuta* aligned to the version 2 genome 

## date: 20220808

## Stringtie assembly

#### from the local directory, copy the reference genome to the remote directory

```
scp Pocillopora_acuta_HIv2.genes.gff3.gz laurenzane@ssh3.hac.uri.edu:/data/putnamlab/shared/Express_Compare/REF
```

#### symbolically link the reference genome (GFF3 format) to the directory with downsampled BAM files 

```
ln -s /data/putnamlab/shared/Express_Compare/REF/Pocillopora_acuta_HIv2.genes.gff3

```
#### Stringtie assembly script 

```
#!/bin/bash
#SBATCH --job-name="ZaneStringtieAssemblyPAcutaV2"
#SBATCH -t 100:00:00
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=laurenzane@uri.edu
#SBATCH -D /data/putnamlab/shared/Express_Compare/stringtie_assembly/Pacuta_BAM
#SBATCH --exclusive #SBATCH -c 20

module load StringTie/2.1.3-GCC-8.3.0

ids=("1041" "1471" "1637" "Sample1" "Sample2" "Sample3")
for file in "${ids[@]}"
do
stringtie -A gene_abundance/"$file".gene_abundance.tab -p 8 --rf -e -G Pocillopora_acuta_HIv2.genes.gff3 -o "$file".gtf "$file".downsampled.bam
done
```

#### arguments used from [Stringtie Documentation](http://ccb.jhu.edu/software/stringtie/index.shtml?t=manual)
    * -A: Gene abundances will be reported (tab delimited format) in the output file with the given name
    * -p: Specify the number of processing threads (CPUs) to use for transcript assembly. The default is 1
    * --rf: Assumes a stranded library fr-firststrand
    * -e: this option directs StringTie to operate in expression estimation mode; this limits the processing of read alignments to estimating the coverage of the transcripts given with the -G option (hence this option requires -G)
    * -G: Use a reference annotation file (in GTF or GFF3 format) to guide the assembly process. The output will include expressed reference transcripts as well as any novel transcripts that are assembled. This option is required by options -B, -b, -e, -C (see below)
    * -o: Sets the name of the output GTF file where StringTie will write the assembled transcripts. This can be specified as a full path, in which case directories will be created as needed. By default StringTie writes the GTF at standard output

#### sbatch information

date | batch number | successful | slurm ouput deleted? | notes
---- | ------------ | ---------- | ----- | --------
20220808 | 169756 | no | yes | wrong genome version 
20220808 | 169761 | yes | no | successful, gene abundance and GTF output moved to stringtie Pacuta Output 

## Stringtie merge 
#### date: 20220809

[helpful website](https://rnabio.org/module-05-isoforms/0005/04/01/Transcript_Assembly_Merge/)
* merge mode merges predicted transcripts generated by Stringtie into a unified transcriptome 
* This mode is used in the new differential analysis pipeline to generate a global, unified set of transcripts (isoforms) across multiple RNA-Seq samples


#### symbolically link gene annotations to directory with gtf files 
```
ln -s /data/putnamlab/shared/Express_Compare/REF/Pocillopora_acuta_HIv2.genes.gff3
```

#### create a .txt file containing a list of gtf files used in merge mode 
```
nano pacuta-v2-merge-list.txt
```

#### copy list of .gtf files into the .txt file with one file per line 
```
1041.gtf
1471.gtf
1637.gtf
Sample1.gtf
Sample2.gtf
Sample3.gtf
```

#### execute Stringtie merge in directory containing .gtf files from Stringtie assembly
```
module load StringTie/2.1.3-GCC-8.3.0
./stringtie --merge -p 8 -G Pocillopora_acuta_HIv2.genes.gff3 -o pacuta-v2-merged.gtf pacuta-v2-merge-list.txt
```
#### Stringtie merge arguments:
    * --merge: specifies merge mode (see above for explanation)
    * -p: specifies the number of processing threads (CPUs) to use for transcript assembly. The default is 1 
    * -G: reference genome included in merging of GTF files
    * -o: output file name
    * <pacuta-v2-merge-list.txt>: this text file contains the list of GTF files to be merged 

## rerun Stringtie with original arguments but using the merged GTF from Stringtie Merge

## note: all directories and output are designated with a "2" to denote second run of Stringtie assembly
Notes from Stringtie manual:
* for each RNA-Seq sample, run StringTie using the -B/-b and -e options in order to estimate transcript abundances and generate read coverage tables for Ballgown. 
    * The -e option is not required but recommended for this run in order to produce more accurate abundance estimations of the input transcripts. 
    * Each StringTie run in this step will take as input the sorted read alignments (BAM file) obtained in step 1 for the corresponding sample and the -G option with the merged transcripts (GTF file) generated by stringtie --merge in step 3. 
    * Please note that this is the only case where the -G option is not used with a reference annotation, but with the global, merged set of transcripts as observed across all samples. 

#### create additional script for Stringtie run 2 

```
nano pacuta-v2-stringtie2-sh
```
```
#!/bin/bash
#SBATCH --job-name="ZaneStringtieAssembly2PAcutaV2"
#SBATCH -t 100:00:00
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --mail-user=laurenzane@uri.edu
#SBATCH -D /data/putnamlab/shared/Express_Compare/stringtie_assembly/Pacuta_BAM
#SBATCH --exclusive #SBATCH -c 20

module load StringTie/2.1.3-GCC-8.3.0

ids=("1041" "1471" "1637" "Sample1" "Sample2" "Sample3")
for file in "${ids[@]}"
do
stringtie -A gene_abundance/"$file".gene_abundance.tab -p 8 --rf -e -G pacuta-v2-merged.gtf -o "$file".2.gtf "$file".downsampled.bam
done
```

#### sbatch information 

date | batch number | successful | slurm ouput deleted? | notes
---- | ------------ | ---------- | ----- | --------
20220809 | 170057 | yes | no | n/a

## compiling GTF output files into gene and transcript count matrices for run 1 and 2 of Stringtie assembly

#### create a text file containing a list of all samples 
```
nano samplelist.txt
```

.txt file contents
```
1041 /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF1/Pacuta/1041.gtf
1637  /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF1/Pacuta/1637.gtf
1471  /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF1/Pacuta/1471.gtf
Sample1  /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF1/Pacuta/Sample1.gtf
Sample2 /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF1/Pacuta/Sample2.gtf
Sample3 /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF1/Pacuta/Sample3.gtf
```
#### compile GTF output files into gene and transcript count matrices

```
module load StringTie/2.1.3-GCC-8.3.0
prepDE.py -g gene_count_matrix.csv -t transcript_count_matrix.csv -i samplelist1.txt

```

#### create a text file containing a list of all samples 
```
nano samplelist2.txt
```

.txt file contents
```
1041.2 /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF2/Pacuta/1041.2.gtf
1637.2  /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF2/Pacuta/1637.2.gtf
1471.2  /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF2/Pacuta/1471.2.gtf
Sample1.2  /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF2/Pacuta/Sample1.2.gtf
Sample2.2 /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF2/Pacuta/Sample2.2.gtf
Sample3.2 /data/putnamlab/shared/Express_Compare/stringtie_assembly/GTF2/Pacuta/Sample3.2.gtf
```
#### compile GTF output files into gene and transcript count matrices

```
module load StringTie/2.1.3-GCC-8.3.0
prepDE.py -g gene_count_matrix2.csv -t transcript_count_matrix2.csv -i sample_list2.txt

```

#### move counts matrices to local directory to be used in DESeq2 R script 
* navigate to your local directory and copy files *from* remote directory to local
```
scp laurenzane@ssh3.hac.uri.edu:/data/putnamlab/shared/Express_Compare/stringtie_assembly/stringtie_counts_matrices/Pacuta/Version2/*.csv* ./
```


